### DISCUSSION TITLE
Using a schema

### DISCUSSION THREAD

**Discussion by avico78** [lk-geimfari/mimesis#1046] [Q&A]
first - amazing package ,working with a few days 
and I love it.
well done guys!

**Questions regarding schema:**
https://mimesis.readthedocs.io/getting_started.html#schema-and-fields

In my schema , i have a sequence fileds which hold a running a uniq sequences,
1.Is there a way for generating number of json automatically?
2.Case yes, is there a way declarer on specific fields as sequence with starting value?

**Comment by lk-geimfari** [comment]
@avico78 Hi! Thank you! Can you show me what do you want, but in code (or pseudo-code), please?

**Reply by avico78** [reply]
Thanks @lk-geimfari 


lets say I have scheme similar to this:
``` python

description = (
     lambda: {
         'sequence_1':   <<this is running senqeunce_start_with_value1>>,
         'sequence_2': <<this is running senqeunce_start_with_value2>>,
          }
 )

```

Meaning sequences fields  has some starting integer values (with default increment by 1 ),
the expected result would be running sequences .

I tried various approach but none was working , I'm sure there's a build in functionality .
but couldn't implement that .
So i tried working with the custom provider .
high level flow:

add a Class which hold sequence name and starting value ,
and return the increment value and keep the current value.
Please consider, single json may have more than one sequence with different starting point,




Really pseudo code  - custom class i wrote doesn't work for multiple sequences.

```python 


## here  define some new SeqProd
class SeqProvider():
    class_counter= 0
    def __init__(self, name, position,increment_by=1):
        self.name = name
        self.position = position
        self.id= SeqProvider.class_counter
        SeqProvider.class_counter += increment_by

generic.add_provider(SeqProvider)


description = (
     lambda: {
         'sequence_1':  _(generic.SeqProvider("sequence_1",10,1),
         'sequence_2':  _(generic.SeqProvider("sequence_2",100,2),
          }
 )
schema = Schema(schema=description)
new_list = schema.create(iterations=2)

```
Expected result:
[{'sequence_1':  10, 'sequence_2': 100}, {'sequence_1':  11, 'sequence_2': 102} ]

**Reply by avico78** [reply]
@lk-geimfari would really appreciate your help here,
if something not clear let me know and i try to provide more details.

Thansk!

**Reply by lk-geimfari** [reply]
Hi! At this moment Mimesis does not support such a feature out the box, sorry.

**Reply by avico78** [reply]
@lk-geimfari ,thanks

a running sequence type of field seems very useful in many cases, 
will u consider to add this functionality?

**Reply by lk-geimfari** [reply]
Sure!
